### PR TITLE
Add Balancing Benchmark Files

### PR DESCRIPTION
**Pull request by manuelkran**
Added onnx-model, system equations and a short description of the proposed balancing model.

### CODE REVIEW

**Comment by ttj** [comment]
Could you please resolve conflict in readme then I will merge? There were updates to the table

**Comment by manuelkran** [comment]
> Could you please resolve conflict in readme then I will merge? There were updates to the table

I updated the readme table, conflict is not resolved.

File: benchmarks/Balancing/dynamics.m [hunk 1]
```diff
@@ -0,0 +1,12 @@
+function dx = dynamics(x, f)
+
+% Cartpole Swingup, 4 states x and the input f (action of the controller)
+% The controller takes (x(1), x(2), x(3), x(4)) as input, its output can
+% be used directly.
+
+dx(1,1) = x(2);
+dx(2,1) = 2*f;
+dx(3,1) = x(4);
+dx(4,1) = (0.08*0.41*(9.8*sin(x(3))-dx(2,1)*cos(x(3)))-0.0021*x(4))/0.0105;
```
Comment (schillic) [inline benchmarks/Balancing/dynamics.m:10]: @manuelkran: 
1. I find it slightly confusing to see `dx(2,1)` on the right-hand side. I suggest to write `2*f`.
2. Do you have a reference/source for the dynamics equations? These are not exactly the equations I know.
3. Can we call the model `CartPole`, which is the common name?

File: benchmarks/Balancing/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+This benchmark is proposed for the ARCH Friendly Competition 2024.
+
+## Benchmark
+
+We consider a pendulum (pole) mounted on a movable cart. The cart can be moved by a controller. The carts postition x1, its velocity x2, the angle of the pole x3 (with 0, 2*pi being the upright postion) and its angular velocity x4 define the state vector of the 4-dimensional system. The system starts in a middle postition of the cart, with the pendulum in the upright position. The controllers goal is to counteract slight deviations in the starting values and balance the pendulum in the middle of the track.
+
+## Specifications and Dynamics
+
+The system dynamics can be found in ```dynamics.m```. The safe states are defined as a stable upright position, which has to be reached after 8 seconds and has to be hold for at least 2 seconds. The controllers step size is 0.02 seconds.
```
Comment (schillic) [inline benchmarks/Balancing/README.md:11]: I find it slightly ambiguous with the text in `specifications.txt` when the counter for the 2 seconds starts. I presume it is version 1, but want to double-check:
1. The counter starts at $t = 8$ (i.e., we have to run until $t = 10$).
2. The counter starts as soon as the "upright position" is reached.

File: benchmarks/Balancing/dynamics.m (hunk 1 above)
Comment (lukaslinauer) [reply benchmarks/Balancing/dynamics.m:10]: Ad 1. & 3.:
I agree to change this. I will replace dx(2,1) with 2*f in the right hand side of dx(4,1) and rename the benchmark to CartPole.

Ad 2. :
The dynamics equations describe a somewhat different, simpler CartPole system (that actually, physically exists), where the cart itself is not influenced by the pole movement.
This is due to the cart position being controlled separately, which allows us to directly set the cart acceleration.

Here is a sketch of the derivation:
  
image

Please let me know if something is unclear about that.

File: benchmarks/Balancing/README.md (hunk 2 above)
Comment (lukaslinauer) [reply benchmarks/Balancing/README.md:11]: You are right, version 1 was intended.

**Pull request merged**